tests/unit/handler.unitTest.ts

```diff
@@ -164,6 +164,66 @@ describe("The lambda function handler", () => {
         expect(updateEuVehicleMock.mock.calls[0][1]).toEqual("m1");
       });
 
+      it("should call the /vehicles/add-provisional function with correct event payload", async () => {
+        const payload = {
+          msUserDetails: {
+            msUser: "dorel",
+            msOid: "12314234"
+          },
+          vin: "XMGDE02FS0H012345",
+          systemNumber: mockData[26].systemNumber,
+          techRecord: mockData[26].techRecord
+        };
+
+        const vehicleRecordEvent = {
+          path: "/vehicles/add-provisional/10000027",
+          pathParameters: {
+            systemNumber: "10000027"
+          },
+          resource: "/vehicles/add-provisional/{systemNumber}",
+          httpMethod: "POST",
+          body: JSON.stringify(payload)
+        };
+        const addProvisionalMock = jest.fn().mockImplementation(() => {
+          return Promise.resolve(new HTTPResponse(200, {}));
+        });
+        TechRecordsService.prototype.addProvisionalTechRecord = addProvisionalMock;
+        const result = await handler(vehicleRecordEvent, ctx);
+        expect.assertions(2);
+        expect(result.statusCode).toEqual(200);
+        expect(TechRecordsService.prototype.addProvisionalTechRecord).toHaveBeenCalled();
+      });
+
+      it("should call the /vehicles/archive function with correct event payload", async () => {
+        const payload = {
+          msUserDetails: {
+            msUser: "dorel",
+            msOid: "12314234"
+          },
+          vin: "XMGDE02FS0H012345",
+          systemNumber: mockData[26].systemNumber,
+          techRecord: mockData[26].techRecord
+        };
+
+        const vehicleRecordEvent = {
+          path: "/vehicles/archive/10000027",
+          pathParameters: {
+            systemNumber: "10000027"
+          },
+          resource: "/vehicles/archive/{systemNumber}",
+          httpMethod: "PUT",
+          body: JSON.stringify(payload)
+        };
+        const archiveTechRecordStatusMock = jest.fn().mockImplementation(() => {
+          return Promise.resolve(new HTTPResponse(200, {}));
+        });
+        TechRecordsService.prototype.archiveTechRecordStatus = archiveTechRecordStatusMock;
```

Comment:
Same for that I think
or we could probably use something like jestGenMockFromModule
https://jestjs.io/docs/en/jest-object#jestgenmockfrommodulemodulename